crontools/crontab.py

```diff
@@ -141,7 +141,8 @@ def iter(self, start_from: Optional[int] = None) -> Iterator[int]:
         step = 1 if self.step is None else self.step
 
         if start_from is not None:
-            begin = max(begin, start_from)
+            while begin < start_from:
```

Comment:
@gilles-crealp  It's correct but in some cases could take a lot of iterations to find the first greater value. For example, for `YearRange.fromstr('*').iter(start_from=2070)` it will take 100 iterations.

Reply:
I think the begin value could be found faster like this:
```
steps = math.ceil(max(start_from - begin, 0) / step)
begin = begin + steps * step
```

Reply:
While counterintuitive, it's not faster.

https://perfpy.com/39

Reply:
perfpy result seems weird. I don't know how it measures execution time internally, but timeit says the opposite:

```python
import timeit
from math import ceil

step = 1
begin = 1970
start_from = 4070


def while_impl():
    local_begin = begin
    while local_begin < start_from:
        local_begin += step


def ceil_impl():
    steps = ceil(max(start_from - begin, 0) / step)
    local_begin = begin + steps * step


executions = 10_000
while_result = timeit.timeit(while_impl, number=executions)
ceil_result = timeit.timeit(ceil_impl, number=executions)

print(f"while impl: {while_result}")
print(f"ceil impl: {ceil_result}")

```

output:
```
while impl: 1.648860622
ceil impl: 0.005186627000000055
```

Reply:
Seems like perfpy reuses setup variables between iterations. https://perfpy.com/41.

Reply:
Pfff. Nice job, perfpy! 🤦‍♂️

Sorry, seemed weird. My bad, I should have more investigated.